datasets/rwth_phoenix_weather_2014_t/rwth_phoenix_weather_2014_t.py

```diff
@@ -0,0 +1,118 @@
+  author={Koller, Oscar and Forster, Jens and Ney, Hermann},
+  journal={Computer Vision and Image Understanding},
+  volume={141},
+  pages={108--125},
+  year={2015},
+  publisher={Elsevier}
+}
+"""
+
+_URL = "https://www-i6.informatik.rwth-aachen.de/ftp/pub/rwth-phoenix/2016/phoenix-2014-T.v3.tar.gz"
+
+_HOMEPAGE = "https://www-i6.informatik.rwth-aachen.de/~koller/RWTH-PHOENIX-2014-T/"
+
+
+class RWTHPhoenixWeather2014T(datasets.GeneratorBasedBuilder):
+    """RWTH-PHOENIX-Weather 2014 T: Parallel Corpus of Sign Language Video, Gloss and Translation"""
+
+    VERSION = datasets.Version("1.0.0")
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=datasets.Features({
+                # sequence of frames
+                "video": datasets.features.Sequence(datasets.features.Array3D(shape=(260, 210, 3), dtype="uint8")),
+                "signer": datasets.Value("string"),  # signer ID
+                "gloss": datasets.Value("string"),  # German sign language gloss
+                "text": datasets.Value("string")  # German translation
+            }),
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+
+        dl_dir = dl_manager.download_and_extract(_URL)
+        base_path = os.path.join(dl_dir, "PHOENIX-2014-T-release-v3", "PHOENIX-2014-T")
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"base_path": base_path, "split": "train"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"base_path": base_path, "split": "dev"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"base_path": base_path, "split": "test"},
+            ),
+        ]
+
+    def _generate_examples(self, base_path, split):
+        """ Yields examples. """
+
+        filepath = os.path.join(base_path, "annotations", "manual", "PHOENIX-2014-T." + split + ".corpus.csv")
+        images_path = os.path.join(base_path, "features", "fullFrame-210x260px", split)
+
+        with open(filepath, "r", encoding="utf-8") as f:
+            data = csv.DictReader(f, delimiter="|", quoting=csv.QUOTE_NONE)
+            for row in data:
+                frames_path = os.path.join(images_path, row["video"])[:-7]
+                np_frames = []
+                for frame_name in os.listdir(frames_path):
+                    frame_path = os.path.join(frames_path, frame_name)
+                    im = Image.open(frame_path)
+                    np_frames.append(np.asarray(im))
+                    im.close()
+
+                yield row["name"], {
+                    "video": np_frames,
```

Comment:
Now that video is just a sequence of np arrays, loading is killed after 170 examples exactly.
I checked, and I can indeed load more than 170 examples at a time, so I suspect something to do with how images are saved in arrow? Not sure. @lhoestq sorry to bother you with this, but do you have any idea?